tests/functional/legacy_api/test_xmlrpc.py

```diff
@@ -34,15 +34,15 @@ def test_xmlrpc_succeeds(app_config, webtest, metrics):
 def test_invalid_arguments(app_config, webtest):
     with pytest.raises(
         xmlrpc.client.Fault,
-        match=r"client error; \('package_name',\): field required",
+        match=r"client error; \('package_name',\): Missing required argument",
     ):
         webtest.xmlrpc("/pypi", "package_releases")
 
 
 def test_arguments_with_wrong_type(app_config, webtest):
     with pytest.raises(
         xmlrpc.client.Fault,
-        match=r"client error; \('serial',\): value is not a valid integer",
+        match=r"client error; \(1,\): Input should be a valid integer",
```

Comment:
Not exactly sure how to get the field names back here.

Reply:
Guessing this is something lost in the python->rust->python cycle introduced in Pydantic 2.x.

Here's a gruesome diff to restore it if we care:

```diff
diff --git a/tests/functional/legacy_api/test_xmlrpc.py b/tests/functional/legacy_api/test_xmlrpc.py
index cd3aceedb..65452831a 100644
--- a/tests/functional/legacy_api/test_xmlrpc.py
+++ b/tests/functional/legacy_api/test_xmlrpc.py
@@ -34,7 +34,7 @@ def test_xmlrpc_succeeds(app_config, webtest, metrics):
 def test_invalid_arguments(app_config, webtest):
     with pytest.raises(
         xmlrpc.client.Fault,
-        match=r"client error; \('package_name',\): Missing required argument",
+        match=r"client error; package_name: Missing required argument",
     ):
         webtest.xmlrpc("/pypi", "package_releases")
 
@@ -42,7 +42,7 @@ def test_invalid_arguments(app_config, webtest):
 def test_arguments_with_wrong_type(app_config, webtest):
     with pytest.raises(
         xmlrpc.client.Fault,
-        match=r"client error; \(1,\): Input should be a valid integer",
+        match=r"client error; serial: Input should be a valid integer",
     ):
         webtest.xmlrpc("/pypi", "changelog_since_serial", "wrong!")
 
@@ -51,7 +51,7 @@ def test_multiple_garbage_types(app_config, webtest):
     with pytest.raises(
         xmlrpc.client.Fault,
         match=(
-            r"client error; \(1,\): Input should be a valid integer; \(2,\): "
+            r"client error; since: Input should be a valid integer; with_ids: "
             r"Input should be a valid boolean"
         ),
     ):
diff --git a/warehouse/legacy/api/xmlrpc/views.py b/warehouse/legacy/api/xmlrpc/views.py
index f826e3094..8b0842af9 100644
--- a/warehouse/legacy/api/xmlrpc/views.py
+++ b/warehouse/legacy/api/xmlrpc/views.py
@@ -17,6 +17,7 @@ import xmlrpc.client
 import xmlrpc.server
 
 from collections.abc import Mapping
+from inspect import signature
 
 from packaging.utils import canonicalize_name
 from pydantic import StrictBool, StrictInt, StrictStr, ValidationError, validate_call
@@ -216,7 +217,13 @@ class TypedMapplyViewMapper(MapplyViewMapper):
             validate_call(fn)(*args, **kwargs)
         except ValidationError as exc:
             raise XMLRPCInvalidParamTypes(
-                "; ".join([f"{e['loc']}: {e['msg']}" for e in exc.errors()])
+                "; ".join(
+                    [
+                        f"{list(signature(fn).parameters.keys())[e['loc'][0]] if isinstance(e['loc'][0], int) else e['loc'][0]}: "
+                        f"{e['msg']}"
+                        for e in exc.errors()
+                    ]
+                )
             )
 
         return super().mapply(fn, args, kwargs)
```

Reply:
Gross, I love it.